src/testdir/test_popup.vim

```diff
@@ -51,16 +51,15 @@ func Test_popup_complete()
   call assert_equal(["August"], getline(1,2))
   %d
 
-  " <BS> - Delete one character from the inserted text (state: 1)
-  " TODO: This should not end the completion, but it does.
-  " This should according to the documentation:
-  " January
-  " but instead, this does
-  " Januar
-  " (idea is, C-L inserts the match from the popup menu
-  " but if the menu is closed, it will insert the character <c-l>
+  " C-L inserts the match from the popup menu
   call feedkeys("aJ\<f5>\<bs>\<c-l>\<esc>", 'tx')
-  call assert_equal(["Januar"], getline(1,2))
+  call assert_equal(["January"], getline(1,2))
+  %d
+
+  " C-L and a <bs> back to `J`, and then select next item from completion
+  " menu which should still be open
+  call feedkeys("aJ\<f5>\<bs>\<bs>\<bs>\<bs>\<bs>\<bs>\<c-n>\<c-y>\<esc>", 'tx')
```

Comment:
Is it possible to add some more test cases to check for edge cases?

Reply:
Sure, but do you have any edge cases in mind?

Reply:
Single and multiple bs behaviors. otherwise LGTM

Reply:
Isn't that what I already have? I changed the failing test to a single bs, and then `<c-l>` to accept it, and then this with multiple bs

Reply:
Yes, it's just that the few case always feels a little loose. but it's okay. It's reasonable.